p2p.go

```diff
@@ -674,6 +679,23 @@ func (h *Host) ConnectedPeers() []core.PeerAddrInfo {
 	return h.peerManager.ConnectedPeers()
 }
 
+// ConnectedPeersByBroadcastTopic returns the connected peers' addrinfo that have subscribed the topic
+func (h *Host) ConnectedPeersByBroadcastTopic(topic string) []core.PeerAddrInfo {
+	var peers []core.PeerAddrInfo
+	if h.pubsub == nil {
+		return nil
+	}
+	topicPeerIDs := h.pubsub.ListPeers(topic)
+	for _, p := range h.peerManager.ConnectedPeers() {
+		if slices.ContainsFunc(topicPeerIDs, func(e peer.ID) bool {
+			return e == p.ID
+		}) {
+			peers = append(peers, p)
+		}
+	}
+	return peers
+}
+
```

Comment:
can you add the test we discussed? A join 3 topics, while B, C, D join 1 topic respectively